src/overlays/actors/ovl_En_Rr/z_en_rr.c

```diff
@@ -1,15 +1,77 @@
+/*
+ * File: z_en_rr.c
+ * Overlay: ovl_En_Rr
+ * Description: Like Like
+ */
+
 #include "z_en_rr.h"
+#include "vt.h"
 
 #define FLAGS 0x00000435
 
 #define THIS ((EnRr*)thisx)
 
+#define RR_MESSAGE_SHIELD (1 << 0)
+#define RR_MESSAGE_TUNIC (1 << 1)
+#define RR_MOUTH 4
+#define RR_BASE 0
+
+typedef enum EnRrReachState {
+    REACH_NONE,
+    REACH_EXTEND,
+    REACH_STOP,
+    REACH_OPEN,
+    REACH_GAPE,
+    REACH_CLOSE
+} EnRrReachState;
+
+typedef enum {
+    RR_DAMAGE_STUN = 1,
+    RR_DAMAGE_FIRE,
+    RR_DAMAGE_ICE,
+    RR_DAMAGE_LIGHT_MAGIC,
+    RR_DAMAGE_LIGHT_ARROW = 11,
+    RR_DAMAGE_UNK_ARROW_1,
+    RR_DAMAGE_UNK_ARROW_2,
+    RR_DAMAGE_UNK_ARROW_3,
+    RR_DAMAGE_NORMAL
+} EnRrDamageEffect;
```

Comment:
Imo this general enum doesn't belong in the file, and should be kept for a later set of documentation changes where the enum is applied to the whole codebase.

Reply:
Damage effect indices aren't general. Each actor assigns what they do individually.

Reply:
I understand that actors define their own damage tables, but wouldn't the damage effects themselves be a universal enum?

I checked other decompiled actors and at least it looks like 1 is always used for stun and 2 is always used for fire/burn, so I imagine that's not a coincidence. Effect 0 seems to be consistently "none" as well btw.

Reply:
Leever uses 11, 12, and 13 for stun, 14 for normal damage, and 3 for ice damage. Redead uses 14 for burning, 1 and 13 for stun, and possibly intended 6 for ice. Guay and Octorok use 0 for normal damage. Shell blade uses 1 for hookshot and 13-15 for various other kinds of damage. I haven't looked through Iron Knuckle or Gossip Stone, but their damage effects are all over the place.